docs/Concepts.md

```diff
@@ -169,6 +199,66 @@ Creating a `TaskRun` will invoke a [Task](#task), running all of the steps until
 completion or failure. Creating a `TaskRun` will require satisfying all of the input
 requirements of the `Task`.
 
+`TaskRun` definition includes `inputs`, `outputs` for `Task` referred in spec.
+
+Input resource includes name and reference to pipeline resource and optionally `paths`. `paths` will be used by `TaskRun` as the resource's new source paths i.e., copy the resource from specified list of paths. `TaskRun` expects the folder and contents to be already present in specified paths. `paths` feature could be used to provide extra files or altered version of existing resource before execution of steps.
```

Comment:
ah great, I get it now! thanks for the example above and the extra detail in the docs :D